doc/Overview/Mvux/Advanced/Commands.md

```diff
@@ -0,0 +1,209 @@
+---
+uid: Overview.Mvux.Advanced.Commands
+---
+
+# Commands
+
+Commands are a way to decouple the UI from the code that performs an action. This makes it easier to maintain and test your code, and it also makes your UI more flexible and adaptable.
+
+A command is an object that implements the [`ICommand`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand) interface. This interface has two methods:
+
+- [`Execute`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand.execute): This method is called when the command is executed.
+- [`CanExecute`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand.canexecute): This method is invoked when the UI needs to determine if the command can be executed. It returns a boolean value indicating if the command can be executed.
+
+Commands can be bound to UI elements. To do this, the `Command` property of the UI element can be used. For example, the following code binds the `Click` event of a button to the `Save` command:
+
+```
+<Button Command="Save">Save</Button>
+```
+
+## Asynchronous commands
+
+The MVUX [`IAsyncCommand`](https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Reactive/Presentation/Commands/IAsyncCommand.cs) interface, is a Command that implements [`ICommand`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.icommand?view=windows-app-sdk-1.3) and adds support for asynchronous operations.  
+As it implements `ICommand`, it can be bound from the View as a regular Command (e.g. in a `Button.Command` property). The advantage is that it can be invoked asynchronously.  
+In addition, it also implements [`INotifyPropertyChanged`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged), to enable tracking if its properties have changed, and [`ILoadable`](xref:Toolkit.Controls.LoadingView#iloadable) - an Uno interface that provides information of an object's state whether it's currently in execution mode or not, and notifies subscribers when this state changes.
+
+![A class diagram of System.ComponentModel.ICommand inheritence structure](../Assets/Commands-2.jpg)
+
+## Creating commands
+
+There are several methods of how to create an MVUX command.
+
+1. Using code generation
+
+    ### Basic commands
+
+    When creating a method in the Model, it will be generated as an `AsyncCommand` in the Bindable Model.
+
+    For example, when declaring a method of the following signatures:
+
+    ```csharp
+    public void DoWork();
+    public ValueTask DoWork();
+    public ValueTask DoWork(CancellationToken ct);    
+    ```
+
+    A `DoWork` command will be generated in the Bindable Proxy Model. When that command is executed via a button-click etc., the method in the Model will be called:
```

Comment:
Will it generate just one method in the BPM? Or 3?

Or is this saying that a `DoWork` command will be generated if **any** of the above method signatures are used?